### PR TITLE
hotfix: Iterate on all available tags to check if tag exist

### PR DESCRIPTION
**Pull request by toninis**
#### Summary

When images are built without `provenance=false` spinwick cannot validate the manifest. 
The library we are using does not support all OCI manifests so we are switching to simple tag checking which works every time.

#### Ticket Link

https://mattermost.atlassian.net/browse/CLD-5172

#### Related Issues
https://github.com/docker/buildx/issues/1533
https://github.com/fraunhoferfokus/deckschrubber/issues/43

#### Release Note

```release-note

```

### CODE REVIEW

**Review by mvitale1989 — APPROVED** [review]
Thanks Antonis! LGTM as a hotfix, although the tags number will only ever grow as of now (`mattermostdevelopment/mm-ee-test` currently has ~1825 tags pushed over the course of ~3 months, so a ~600 tags/month increase), so perhaps it would still be more sustainable to eventually switch to a modern library, and directly retrieve the image manifest again.

**Comment by toninis** [comment]
> Thanks Antonis! LGTM as a hotfix, although the tags number will only ever grow as of now (`mattermostdevelopment/mm-ee-test` currently has ~1825 tags pushed over the course of ~3 months, so a ~600 tags/month increase), so perhaps it would still be more sustainable to eventually switch to a modern library, and directly retrieve the image manifest again.

Agreed . The api is really fast in any case . Should we establish a garbage collection policy ?

**Comment by mvitale1989** [comment]
> Agreed . The api is really fast in any case . Should we establish a garbage collection policy ?

No urgent need for this PR's purpose I think. But I would open a ticket in our backlog to have a discussion about this topic in general.

File: server/builds.go [hunk 1]
```diff
@@ -48,14 +48,16 @@ func (b *Builds) waitForImage(ctx context.Context, s *Server, reg *registry.Regi
 		case <-ctx.Done():
 			return pr, errors.New("timed out waiting for image to publish")
 		case <-time.After(30 * time.Second):
-			_, err := reg.ManifestDigest(imageToCheck, desiredTag)
+			availableTags, err := reg.Tags(imageToCheck)
```
Comment (spirosoik) [inline server/builds.go:51]: wondering if there is a test. if it's not it's ok.

**Review by spirosoik — APPROVED** [review]
thanks for fixing this

**Comment by toninis** [comment]
Closing this as it is not needed . Once we introduce multi architecture images we will re-evaluate

**Pull request closed without merging**